mne/preprocessing/tests/test_ica.py

```diff
@@ -866,4 +871,42 @@ def test_n_components_and_max_pca_components_none(method):
     assert_is_none(ica.n_components)
 
 
+@requires_sklearn
+@testing.requires_testing_data
+def test_ica_ctf():
+    """Test run ICA computation on ctf data with/without compensation."""
+    method = 'fastica'
+    raw = read_raw_ctf(ctf_fname, preload=True)
+    events = make_fixed_length_events(raw, 99999)
+    for comp in [0, 1]:
+        raw.apply_gradient_compensation(comp)
+        epochs = Epochs(raw, events, None, -0.2, 0.2, preload=True)
+        evoked = epochs.average()
+
+        # test fit
+        for inst in [raw, epochs]:
+            ica = ICA(n_components=2, random_state=0, method=method)
+            with warnings.catch_warnings(record=True):  # convergence
+                ica.fit(raw)
+
+        # test apply and get_sources
+        for inst in [raw, epochs, evoked]:
+            ica.apply(inst)
+            ica.get_sources(inst)
+
+    # test mixed compensation case
+    raw.apply_gradient_compensation(0)
+    ica = ICA(n_components=2, random_state=0, method=method)
+    with warnings.catch_warnings(record=True):  # convergence
+        ica.fit(raw)
+    raw.apply_gradient_compensation(1)
+    epochs = Epochs(raw, events, None, -0.2, 0.2, preload=True)
+    evoked = epochs.average()
+    for inst in [raw, epochs, evoked]:
+        with pytest.raises(RuntimeError, match='Compensation grade of ICA'):
+            ica.apply(inst)
+        with pytest.raises(RuntimeError, match='Compensation grade of ICA'):
+            ica.get_sources(inst)
```

Comment:
How long does this take to run? Can we set the number of iterations to some tiny number (1 or 2) to make it faster, since we don't actually care if the decomp is any good?

Reply:
`test_ica::test_ica_ctf` takes 6 seconds with `max_iter=2` and 12 seconds without the default `max_iter`

Reply:
That's a good speedup, let's add it